hudi-client/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -274,6 +275,7 @@ public HoodieIndexConfig build() {
           DEFAULT_GLOBAL_SIMPLE_INDEX_PARALLELISM);
       setDefaultOnCondition(props, !props.containsKey(SIMPLE_INDEX_UPDATE_PARTITION_PATH),
           SIMPLE_INDEX_UPDATE_PARTITION_PATH, DEFAULT_SIMPLE_INDEX_UPDATE_PARTITION_PATH);
+      setDefaultOnCondition(props, !isHBaseIndexConfigSet, HoodieHBaseIndexConfig.newBuilder().fromProperties(props).build());
```

Comment:
@yanghua Maybe this can be done specially for HBase index. But I'm also open to leave it as is: any sub-config should be initialized regardless.

```suggestion
      setDefaultOnCondition(props, indexType==IndexType.HBASE && !isHBaseIndexConfigSet, HoodieHBaseIndexConfig.newBuilder().fromProperties(props).build());
```